packages/@react-aria/landmark/src/useLandmark.ts

```diff
@@ -0,0 +1,261 @@
+      } else {
+        let labels = [...landmarksWithRole].map(landmark => landmark.label);
+        let duplicateLabels = labels.filter((item, index) => labels.indexOf(item) !== index);
+  
+        duplicateLabels.forEach((label) => {
+          console.warn(`Page contains more than one landmark with the '${role}' role and '${label}' label. If two or more landmarks on a page share the same role, they must have unique labels.`);
+        });
+      }
+    }
+  }
+
+  /**
+   * Get the landmark that is the closest parent in the DOM.
+   * Returns undefined if no parent is a landmark.
+   */
+  private closestLandmark(element: HTMLElement) {
+    let landmarkMap = new Map(this.landmarks.map(l => [l.ref.current, l]));
+    let currentElement = element;
+    while (!landmarkMap.has(currentElement) && currentElement !== document.body) {
+      currentElement = currentElement.parentElement;
+    }
+    return landmarkMap.get(currentElement);
+  }
+
+  /**
+   * Gets the next landmark, in DOM focus order, or previous if backwards is specified.
+   * If nested, next should be the child landmark. 
+   * If last landmark, next should be the first landmark.
+   * If not inside a landmark, will return first landmark.
+   * Returns undefined if there are no landmarks.
+   */
+  public getNextLandmark(element: HTMLElement, {backward}: {backward?: boolean }) {
+    if (this.landmarks.length === 0) {
+      return undefined;
+    }
+    
+    let currentLandmark = this.closestLandmark(element);
+    let nextLandmarkIndex = backward ? -1 : 0;
+    if (currentLandmark) {
+      nextLandmarkIndex = this.landmarks.findIndex(landmark => landmark === currentLandmark) + (backward ? -1 : 1);
+    }
+
+    // Wrap if necessary
+    if (nextLandmarkIndex < 0) {
+      nextLandmarkIndex = this.landmarks.length - 1;
+    } else if (nextLandmarkIndex >= this.landmarks.length) {
+      nextLandmarkIndex = 0;
+    }
+
+    return this.landmarks[nextLandmarkIndex];
+  }
+
+  /**
+   * Look at next landmark. If an element was previously focused inside, restore focus there.
+   * If not, focus the first focusable element inside the lanemark.
+   * If no focusable elements inside, go to the next landmark.
+   * If no landmarks at all, or none with focusable elements, don't move focus.
+   */
+  public f6Handler(e: KeyboardEvent) {
+    if (e.key === 'F6') {
+      e.preventDefault();
+      e.stopPropagation();
+
+      let backward = e.shiftKey;
+      let nextLandmark = this.getNextLandmark(e.target as HTMLElement, {backward});
+
+      // If no landmarks, return
+      if (!nextLandmark) {
+        return;
+      }
+
+      // If alt key pressed, focus main landmark
+      if (e.altKey) { 
+        let main = this.getLandmarksByRole('main');
+        if (main.size > 0) {
+          this.focusLandmark([...main][0].ref.current);
+        }
+        return;
+      }
```

Comment:
Should this be focusing the lastFocused within the main landmark?

Reply:
reminder for posterity: team discussed and decided to have it always focus the main landmark regardless if the elements inside were focused before